### PR TITLE
fix bugs when push logic ops and bit ops down

### PR DESCRIPTION
**Pull request by XuHuaiyu**
ast.And is BitAnd, ast.AndAnd is LogicAnd
This commit correct the error check before, and add some test to test whether Logic Operators been pushed down to kv.

@coocood @hanfei1991 @shenli @zimulala @tiancaiamao PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: plan/plan_test.go [hunk 1]
```diff
@@ -273,6 +273,79 @@ func (s *testPlanSuite) TestTopnPushDown(c *C) {
 	}
 }
 
+func (s *testPlanSuite) TestLogicOpsPushDown(c *C) {
```
Comment (shenli) [inline plan/plan_test.go]: Add some comments for this test case.

**Comment by XuHuaiyu** [comment]
PTAL @shenli  @hanfei1991

**Comment by shenli** [comment]
LGTM

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**